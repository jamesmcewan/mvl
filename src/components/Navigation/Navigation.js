import React from 'react';
import styled from 'styled-components';
import { Button } from '..';
import camelCase from 'lodash.camelcase';
import mvl from '../../images/logo.svg';
import { isArray } from 'util';

const StyledNav = styled.nav`
  width: 100%;
  position: fixed;
  z-index: 1;
  bottom: 0;
  background: #151617 linear-gradient(#1a0f21, #151617 70%) no-repeat;
  padding: 0;
  display: flex;
  flex-flow: row wrap;

  @media screen and (min-width: 700px) {
    bottom: auto;
  }
`;

const StyledImg = styled.img`
  max-width: 150px;
`;

const StyledHeader = styled.header`
  flex-grow: 0;
  background: #1e1b2f;
  width: 100%;
  text-align: center;
  order: 1;

  @media screen and (min-width: 700px) {
    order: 0;
  }
`;

const Navigation = ({ dateDescriptor, dateDescriptors, handleClick }) => (
  <StyledNav>
    <StyledHeader>
      <StyledImg src={mvl} alt="MVL" />
    </StyledHeader>
    {isArray(dateDescriptors) &&
      dateDescriptors.map(d => {
        const current = camelCase(d) === dateDescriptor;
        return (
          <Button
            key={`button-${d}`}
            dateDescriptor={d}
            handleClick={handleClick}
            current={current}
          />
        );
      })}
  </StyledNav>
);

export default Navigation;
