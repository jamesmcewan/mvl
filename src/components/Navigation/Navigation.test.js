import React from 'react';
import { mount } from 'enzyme';
import Navigation from './Navigation';
import 'jest-styled-components';

describe('Navigation', () => {
  it('should render without crashing', () => {
    const wrapper = mount(<Navigation />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render the correct JSX', () => {
    const wrapper = mount(<Navigation />);
    expect(wrapper).toMatchSnapshot();
  });
});
