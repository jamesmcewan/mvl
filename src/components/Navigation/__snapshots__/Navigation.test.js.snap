// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render the correct JSX 1`] = `
.c0 {
  width: 100%;
  position: fixed;
  z-index: 1;
  bottom: 0;
  background: #151617 linear-gradient(#1a0f21,#151617 70%) no-repeat;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c2 {
  max-width: 150px;
}

.c1 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  background: #1e1b2f;
  width: 100%;
  text-align: center;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

@media screen and (min-width:700px) {
  .c0 {
    bottom: auto;
  }
}

@media screen and (min-width:700px) {
  .c1 {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
  }
}

<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Navigation>
      <styled.nav>
        <nav
          className="c0"
        >
          <styled.header>
            <header
              className="c1"
            >
              <styled.img
                alt="MVL"
                src="logo.svg"
              >
                <img
                  alt="MVL"
                  className="c2"
                  src="logo.svg"
                />
              </styled.img>
            </header>
          </styled.header>
        </nav>
      </styled.nav>
    </Navigation>
  </Router>
</BrowserRouter>
`;
